src/front/glsl/parser.rs

```diff
@@ -828,9 +828,19 @@ pomelo! {
     translation_unit ::= translation_unit external_declaration;
 
     external_declaration ::= function_definition(f) {
-        let name = f.name.clone();
-        let handle = extra.functions.append(f);
-        if let Some(name) = name {
+        if f.name == extra.entry {
+            let name = extra.entry.take().unwrap();
+            extra.module.entry_points.insert(
+                (extra.shader_stage, name),
+                EntryPoint {
+                    early_depth_test: None,
+                    workgroup_size: [1; 3], //TODO
```

Comment:
Do you still need to do something here, or is it for a later?

Reply:
that matches the current behavior, so there is no need to fix this in this PR. I wanted to leave a note here to address later by somebody.